dizoo/league_demo/league_scheduler.py

```diff
@@ -0,0 +1,172 @@
+        schedule_mode = cfg.policy.scheduler.schedule_mode
+        factor = cfg.policy.scheduler.factor
+        change_range = cfg.policy.scheduler.change_range
+        threshold = cfg.policy.scheduler.threshold
+        optimize_mode = cfg.policy.scheduler.optimize_mode
+        patience = cfg.policy.scheduler.patience
+        cooldown = cfg.policy.scheduler.cooldown
+        
+        assert schedule_parameter in ['entropy_weight', 'learning_rate'], 'The parameter to be scheduled should be one of [\'entropy_weight\', \'learning_rate\']'
+        self.schedule_parameter = schedule_parameter
+        
+        assert schedule_mode in ['reduce','add','multi','div'], 'The schedule mode should be one of [\'reduce\', \'add\', \'multi\',\'div\']'
+        self.schedule_mode = schedule_mode
+        
+        assert isinstance(factor, float) or isinstance(factor, int), 'The factor should be a float/int number greater than 0'
+        assert factor > 0, 'The factor should be a float/int number greater than 0'
+        self.factor = factor
+
+        assert isinstance(change_range, list) and len(change_range) == 2, 'The change_range should be a list with 2 float numbers'
+        assert (isinstance(change_range[0], float) or isinstance(change_range[0], int)) and (isinstance(change_range[1], float) or isinstance(change_range[1], int)), 'The change_range should be a list with 2 float/int numbers'
+        if schedule_parameter == 'entropy_weight':
+            assert (change_range[0] <= cfg.policy.learn.entropy_weight 
+                    and change_range[1] >= cfg.policy.learn.entropy_weight), 'The change range should cover the initial entropy weight value'
+        if schedule_parameter == 'learning_rate':
+            assert (change_range[0] <= cfg.policy.learn.learning_rate 
+                    and change_range[1] >= cfg.policy.learn.learning_rate), 'The change range should cover the initial learning rate value'
+        self.change_range = change_range
+
+        assert isinstance(threshold, float) or isinstance(threshold, int), 'The threshold should be a float/int number'
+        self.threshold = threshold
+
+        assert optimize_mode in ['min', 'max'], 'The optimize_mode should be one of [\'min\', \'max\']'
+        self.optimize_mode = optimize_mode
+
+        assert isinstance(patience, int), 'The patience should be a integer greater than or equal to 0'
+        assert patience >= 0, 'The patience should be a integer greater than or equal to 0'
+        self.patience = patience
+
+        assert isinstance(cooldown, int), 'The cooldown_counter should be a integer greater than or equal to 0'
+        assert cooldown >= 0, 'The cooldown_counter should be a integer greater than or equal to 0'
+        self.cooldown = cooldown
+        self.cooldown_counter = cooldown
+
+        self.last_metrics = None
+        self.bad_epochs_num = 0
+        
+
+    def step(self, metrics, cfg):
```

Comment:
Add python typing lint for the inputs and outputs of function and method

Reply:
done.